aws/resource_aws_directory_service_directory_test.go

```diff
@@ -280,22 +301,21 @@ func testAccCheckDirectoryServiceDirectoryDestroy(s *terraform.State) error {
 			DirectoryIds: []*string{aws.String(rs.Primary.ID)},
 		}
 		out, err := dsconn.DescribeDirectories(&input)
+
+		if isAWSErr(err, "EntityDoesNotExistException", "") {
+			continue
+		}
+
 		if err != nil {
-			// EntityDoesNotExistException means it's gone, this is good
-			if dserr, ok := err.(awserr.Error); ok && dserr.Code() == "EntityDoesNotExistException" {
-				return nil
-			}
 			return err
 		}
 
 		if out != nil && len(out.DirectoryDescriptions) > 0 {
 			return fmt.Errorf("Expected AWS Directory Service Directory to be gone, but was still found")
 		}
-
-		return nil
 	}
 
-	return fmt.Errorf("Default error in Service Directory Test")
```

Comment:
Previously this `CheckDestroy` function was not properly exiting when there were no resources in the state (e.g. creation failed):

```
    testing.go:629: Error destroying resource! WARNING: Dangling resources
        may exist. The full state and error is shown below.

        Error: Check failed: Default error in Service Directory Test

        State: <no state>
```